### PR TITLE
[Search] introduce a new search-react package. 

### PR DESCRIPTION
**Pull request by emmaindal**
## Hey, I just made a Pull Request!

This PR introduces a new search-react package which will hold things the search plugin itself and other frontend plugins (techdocs, home) depend on. For now we have migrated `searchApiRef`, `SearchApi` interface and the search context, to get rid of plugins to depend on other plugins. But expect more reusable components to be migrated here in the near future.

The "DefaultTemplate" storybook example for the home plugin is also moved to the app for now (see https://github.com/backstage/backstage/pull/10752/commits/bc31ff1bbd192a5d21b7b02f63d82536cbb636ca), let me know if you disagree with this and think it has a better home. 

Closes: https://github.com/backstage/backstage/issues/10672

**Todo:**

- [x] Write changesets

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by emmaindal** [comment]
@backstage/maintainers is there a better place to use the default logos from? if so I can delete those and import from another package.

**Comment by emmaindal** [comment]
Thanks @iamEAP and @jhaals yes I think lets get this one out, I'll start more more things over from the search plugin but wanted to unblock as quick as possible before I start move more things. So will definitely get back to the api reports warnings!

File: plugins/search/src/apis.ts
Comment (freben) [reply plugins/search/src/apis.ts:28]: Up to you. I think it's perfectly fine in general to have duplicates, especially if you intend to "freeze in time" the old one. Makes it easier to evolve the new place I think

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
I see a few small remaining warnings in the api reports, but it's not all that consequential to the release so let's get it in and iterate!

File: plugins/search-react/src/context/SearchContext.tsx
Comment (Rugvip) [inline plugins/search-react/src/context/SearchContext.tsx:56]: Would be best if we could add a version bridge for this one right? I'm assuming this is a context that is shared pretty broadly across the app, potentially consumed by different plugins on different versions of `search-react`. Perhaps no need to rush that in today, but best if we could get it in place before 1.1? (@backstage/techdocs-core)

Comment (emmaindal) [reply plugins/search-react/src/context/SearchContext.tsx:56]: Thanks @Rugvip I think I'll try to not export the context itself in the next one (when moving over more compoents) and just the provider + hooks. Do we still need a version bridge? And on that topic, I think the version bridge thing is a new thing to me, haven't heard about it before, where can I read up on it?

Comment (Rugvip) [reply plugins/search-react/src/context/SearchContext.tsx:56]: Yep, the versioned context both makes it easier to evolve a context, but also makes sure that package duplication doesn't break the app. A regular React context is by-reference so it can't handle package duplication. The API itself is reasonably well documented at https://backstage.io/docs/reference/version-bridge, but ofc look at existing usage in the repo just to match error handling and so on a bit.

Comment (emmaindal) [reply plugins/search-react/src/context/SearchContext.tsx:56]: Thank you, will look into it! @Rugvip

Comment (emmaindal) [reply plugins/search-react/src/context/SearchContext.tsx:56]: https://github.com/backstage/backstage/pull/10808

**Pull request merged**
